### PR TITLE
x11-misc/: Fix build with gcc6 and one version bump

### PR DESCRIPTION
**Pull request by Moiman**

### CODE REVIEW

File: x11-misc/evolvotron/evolvotron-0.6.3-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+HOMEPAGE="https://sourceforge.net/projects/evolvotron/"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86 ~x86-fbsd"
+IUSE=""
+
+RDEPEND="
+	dev-libs/boost
```
Comment (SoapGentoo) [inline x11-misc/evolvotron/evolvotron-0.6.3-r1.ebuild]: seeing that you have boost in `RDEPEND`, I assume it links to some boost lib? If so, boost needs `:=`

File: x11-misc/evolvotron/evolvotron-0.6.3-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	dev-qt/qtcore:4
+	dev-qt/qtgui:4
+"
+DEPEND="${RDEPEND}"
+
+S=${WORKDIR}/${PN}
+
+PATCHES=( "${FILESDIR}"/${P}-gcc6.patch )
+
+DOCS=( BUGS NEWS README TODO USAGE )
```
Comment (SoapGentoo) [inline x11-misc/evolvotron/evolvotron-0.6.3-r1.ebuild]: is `USAGE` that important? Does it contain important instructions? If not, I would just nix `DOCS` (your decision though)

File: x11-misc/fbpanel/fbpanel-7.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+RDEPEND="
+	dev-libs/glib:2
+	alsa? ( media-libs/alsa-lib )
+	x11-libs/gdk-pixbuf:2[X]
+	x11-libs/gtk+:2
+	x11-libs/libX11
+"
+DEPEND="
+	${RDEPEND}
+	dev-lang/python:2.7
```
Comment (SoapGentoo) [inline x11-misc/fbpanel/fbpanel-7.0.ebuild]: python cannot be used like this. If python is a build-time dep -> `python-any-r1`, if its also a runtime time -> `python-single-r1`.

File: x11-misc/fbpanel/files/fbpanel-7.0-shebangs.patch [hunk 4]
```diff
@@ -0,0 +1,135 @@
+diff -u -r a/.config/help b/.config/help
+--- a/.config/help	2015-12-05 08:34:19.000000000 +0200
++++ b/.config/help	2017-02-25 15:18:46.328392853 +0200
+@@ -1,4 +1,4 @@
+-#!/usr/bin/python
++#!/usr/bin/env python2
```
Comment (SoapGentoo) [inline x11-misc/fbpanel/files/fbpanel-7.0-shebangs.patch]: :heart: 
Can you try upstreaming this and giving them the rundown why `env` based-lookup is better than hardcoded paths? (https://blogs.gentoo.org/mgorny/2016/02/08/a-quick-note-on-portable-shebangs/)

File: x11-misc/fracplanet/fracplanet-0.4.0-r1.ebuild [hunk 5]
```diff
@@ -25,17 +25,25 @@ DEPEND="${RDEPEND}
 
 S=${WORKDIR}/${PN}
 
-PATCHES=( "${FILESDIR}/${P}-gold.patch" )
+PATCHES=(
+	"${FILESDIR}/${P}-gold.patch"
+	"${FILESDIR}/${P}-gcc6.patch"
+)
+
+HTML_DOCS=( fracplanet.htm fracplanet.css )
```
Comment (SoapGentoo) [inline x11-misc/fracplanet/fracplanet-0.4.0-r1.ebuild]: `HTML_DOCS=( fracplanet.{htm,css} )`

File: x11-misc/piedock/piedock-1.6.6-r1.ebuild [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
+	https://github.com/markusfisch/PieDock/commit/a7fda1896f1cc6966ba0fa8912e9b404c1b0be97.patch -> ${P}-gcc6.patch"
 
 LICENSE="MIT"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
 IUSE="gtk kde"
 
 RDEPEND="
-	media-libs/libpng
+	media-libs/libpng:0
```
Comment (SoapGentoo) [inline x11-misc/piedock/piedock-1.6.6-r1.ebuild]: does it link to libpng (which I strongly would think)? linked libpng deps always require `:0=`

File: x11-misc/xplanet/xplanet-1.3.1-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~hppa ~ppc ~ppc64 ~sparc ~x86"
+IUSE="gif jpeg png tiff truetype X"
+
+RDEPEND="
+	gif? ( media-libs/giflib:= )
+	jpeg? ( virtual/jpeg:* )
```
Comment (SoapGentoo) [inline x11-misc/xplanet/xplanet-1.3.1-r1.ebuild]: you probably want `:0` here

File: x11-misc/xplanet/xplanet-1.3.1-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+	png? (
+		media-libs/libpng:0
```
Comment (SoapGentoo) [inline x11-misc/xplanet/xplanet-1.3.1-r1.ebuild]: `:0=` likely

File: x11-misc/xplanet/xplanet-1.3.1-r1.ebuild [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+	truetype? (
+		media-libs/freetype:2
+		x11-libs/pango
+		)
+	X? (
+		x11-libs/libX11
+		x11-libs/libXScrnSaver
+		x11-libs/libXext
+		x11-libs/libXt
+		)
```
Comment (SoapGentoo) [inline x11-misc/xplanet/xplanet-1.3.1-r1.ebuild]: always have parentheses end on the same column as the USE flag:
```
X? (
	blabla
)
```

File: x11-misc/xplanet/xplanet-1.3.1-r1.ebuild [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+)
+
+src_configure() {
+	# econf says 'checking pnm.h presence... no'
+	use png && append-cppflags -I/usr/include/netpbm
+
+	local myconf
+
+	use X \
+		&& myconf+=" --with-x --with-xscreensaver" \
```
Comment (SoapGentoo) [inline x11-misc/xplanet/xplanet-1.3.1-r1.ebuild]: could you make `myconf` a bash array and use that instead? arrays are less retarded than whitespace separated strings

File: x11-misc/xplanet/xplanet-1.3.1-r1.ebuild [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-giflib.patch
+	"${FILESDIR}"/${P}-remove-null-comparison.patch
+)
+
+src_configure() {
+	# econf says 'checking pnm.h presence... no'
+	use png && append-cppflags -I/usr/include/netpbm
```
Comment (SoapGentoo) [inline x11-misc/xplanet/xplanet-1.3.1-r1.ebuild]: `use png && append-cppflags -I"${EPREFIX}"/usr/include/netpbm`

File: x11-misc/evolvotron/evolvotron-0.6.3-r1.ebuild (hunk 1 above)
Comment (Moiman) [inline x11-misc/evolvotron/evolvotron-0.6.3-r1.ebuild]: I looks like it just needs header file.

File: x11-misc/evolvotron/evolvotron-0.6.3-r1.ebuild (hunk 2 above)
Comment (Moiman) [inline x11-misc/evolvotron/evolvotron-0.6.3-r1.ebuild]: Looks useful: https://sourceforge.net/p/evolvotron/code/HEAD/tree/trunk/USAGE

Comment (SoapGentoo) [inline x11-misc/evolvotron/evolvotron-0.6.3-r1.ebuild]: ok keep it

File: x11-misc/fbpanel/files/fbpanel-7.0-shebangs.patch (hunk 4 above)
Comment (Moiman) [inline x11-misc/fbpanel/files/fbpanel-7.0-shebangs.patch]: Yes. I can try. I was on my mind already.

Comment (Moiman) [inline x11-misc/fbpanel/files/fbpanel-7.0-shebangs.patch]: https://github.com/aanatoly/fbpanel/pull/39

**Pull request closed without merging**